src/jit/morph.cpp

```diff
@@ -7309,7 +7403,10 @@ GenTreePtr          Compiler::fgMorphCall(GenTreeCall* call)
         }
 #endif // FEATURE_TAILCALL_OPT
 
-        fgFixupStructReturn(call);
+        if (varTypeIsStruct(call))
+        {
+            fgFixupStructReturn(call);
+        }
 
```

Comment:
It seem better to have the caller perform this check to me.  
We only call this in two places and to check at the call site it pretty cheap.  
We also don't like to have methods that have early exits.
Although a single early exit at the start is probably OK.